samples/subscriptions.js

```diff
@@ -97,31 +96,36 @@ async function createFlowControlledSubscription(
   /**
    * TODO(developer): Uncomment the following lines to run the sample.
    */
-  // const topicName = 'my-topic';
   // const subscriptionName = 'my-sub';
   // const maxInProgress = 5;
-  // const maxBytes = 10000;
-
-  const topic = pubsub.topic(topicName);
+  // const timeout = 10;
 
-  const options = {
+  const subscriberOptions = {
     flowControl: {
-      maxBytes: maxBytes,
       maxMessages: maxInProgress,
     },
   };
 
-  const subscription = topic.subscription(subscriptionName, options);
+  // References an existing subscription.
+  // Note that flow control settings are not persistent across subscribers.
+  const subscription = pubsub.subscription(subscriptionName, subscriberOptions);
 
-  // Creates a new subscription
-  // Note that flow control configurations are not persistent
-  const [newSubscription] = await subscription.get({
-    autoCreate: true,
-  });
   console.log(
-    `Subscription ${newSubscription.name} created with a maximum of ${maxInProgress} unprocessed messages.`
+    `Subscriber to subscription ${subscription.name} is ready to receive messages at a controlled volume of ${maxInProgress} messages.`
   );
 
+  const messageHandler = function(message) {
+    console.log(`Received message: ${message.id}`);
+    console.log(`\tData: ${message.data}`);
+    message.ack();
+  };
+
+  subscription.on(`message`, messageHandler);
+
+  setTimeout(() => {
+    subscription.removeListener(`message`, messageHandler);
```

Comment:
Alternatively you could call `subscription.close()` to simplify the example.

Reply:
Done.